scipy/signal/tests/test_filter_design.py

```diff
@@ -1705,6 +1705,19 @@ def test_fs_validation(self):
         with pytest.raises(ValueError, match="Sampling.*be none"):
             bilinear(b, a, fs=None)
 
+    def test_trim_zeros(self):
+        # Test for leading zeros
+        b = [0, 0, 1, 2, 3, 0, 0]
+        expected_b = [1, 2, 3]
+        b_trimmed = _trim_zeros(np.array(b), 'fb')
+        assert_array_almost_equal(b_trimmed, expected_b)
+
+        # Test for trailing zeros
+        b = [1, 2, 3, 0, 0]
+        expected_b = [1, 2, 3]
+        b_trimmed = _trim_zeros(np.array(b), 'fb')
+        assert_array_almost_equal(b_trimmed, expected_b)
```

Comment:
If you were comparing floating point arrays I'd suggest switching to `assert_allclose()` as the documentation for `assert_array_almost_equal` suggests.

In this case though, it looks like your expected values are integers? So, `assert_array_equal` for an exact equality check may actually be suitable, or possibly even Evgeni's suggestion `xp_assert_equal` could suit.